### DISCUSSION TITLE
stash.loadData() functionality

### DISCUSSION THREAD

**Discussion by FilipMW** [pactumjs/pactum#258] [Q&A]
Hi, I'm evaluating your tool and I'm trying to load data from `data` folder (within I have a `templates` and `maps` folders, respectively, just like in your [example](https://github.com/pactumjs/pactum-examples/tree/main/data-management/data).

In total, it looks like this:
![image](https://user-images.githubusercontent.com/76940289/210785710-52a104d0-3565-470e-9f30-d0e4dd2399d2.png)

```js
const { request, stash } = require('pactum');

before(() => {
  stash.loadData('tests/IntegrationTests/COAPI/data');
  request.setBaseUrl('$M{environments.dev}');
});
```
The test itself:
```js
const { describe, it } = require('node:test');
const {spec} = require('pactum');

describe('Register new user', () => {

const { request, stash } = require('pactum');

before(() => {
  stash.loadData('tests/IntegrationTests/COAPI/data');
  request.setBaseUrl('$M{environments.dev}');
});

  it('INIT get token', async () => {
    await spec()
      .post('/api/checkout-with-products/products')
      .expectStatus(200)
      .stores('token', 'token')
  });
  
  it('Register user', async () => {
    await spec()
      .post('/api/checkout-with-products/register')
      .withHeaders('API-Token', '$S{token}')
      .withJson({
        '@DATA:TEMPLATE@': 'user'
      })
      .expectStatus(201)
      .expectBodyContains("test@testmail.com")
  });
});
```
`environments.json`:
```json
{ 
  "environments": 
  {
    "dev": "http://localhost",
    "ci": "http://nginx-ci"
  }
}
```
And so this code should in theory load data for further tests and set the base url (`package.json` is in the main project folder). It does neither. Could you please elaborate on the use of `stash.loadData()`? It's not loading anything, no matter the path to `data` folder - even absolute path fails. Adding maps manually in test and setting the baseUrl works, but it's not enough for our use case.

**Comment by ASaiAnudeep** [comment]
Are you facing any errors while using `loadData`?

The path of the data folder should be from the current working dir where you are executing the test run command.

May be try this `stash.loadData('./tests/IntegrationTests/COAPI/data');`

**Reply by FilipMW** [reply]
I'm facing errors connected to invalid URL, like there's no base URL. I've already tried your solution and also tried using absolute path. Both don't assign `localhost` as the base URL.

**Reply by ASaiAnudeep** [reply]
Could you please create a sample repo to replicate the issue and share it with us.
